### DISCUSSION TITLE
windowRules: Firefox vs Thunderbird

### DISCUSSION THREAD

**Discussion by Narrat** [labwc/labwc#1987] [Q&A] [answered]
Since forever I have this strange issue and it finally bugs me enough :D 
I have simple windowRules set for Firefox and Thunderbird, which are essentially the same platform. The identifier in question checked via `lswt`.
But the strange thing is all actions defined for firefox are working, but for thunderbird only one out of three. And I don't understand why. I could see reason if none of the actions would be applied, but not this some may work and some not.

Rules in question:
```xml
<windowRule identifier="*firefox*">
    <ignoreFocusRequest>yes</ignoreFocusRequest>
     <action name="ToggleDecorations" />
     <action name="SendToDesktop" to="2"/>
</windowRule>
<windowRule identifier="*thunderbird*">
    <ignoreFocusRequest>yes</ignoreFocusRequest>
    <action name="ToggleDecorations" />
    <action name="SendToDesktop" to="5"/>
</windowRule>
```
In principal the same structure. Firefox has no decorations, is on workspace 2 and focus requests are ignored.
Thunderbird will appear on workspace 5, but still has decorations and focus requests will bring it into the focus.
The wayland specific env var is set, so both should be wayland native apps.

Can someone reproduce or did I make a mistake and just don't see it?

**Comment by stefonarch** [comment]
You are using server side decorations I think? While testing the snippet with CSD I get:
![immagine](https://github.com/user-attachments/assets/73c9918d-bc0e-4a57-8090-2e2408e5097b)
Except that with SSD `sendtodesktop` and `toggledecorations` are working, but when I click "send link by email" in firefox it switches the desktop and activates TB.

**Reply by Narrat** [reply]
Yes, I'm using SSD.
Funny but even more irritating, that this causes different results if SSD or CSD is active.

**Reply by Consolatis** [reply, marked as answer]
You can use [SetDecorations](https://labwc.github.io/labwc-actions.5.html#entry_action_name=setdecorations_decorations=value_forcessd=no) instead of ToggleDecorations.

**Reply by Narrat** [reply]
> You can use [SetDecorations](https://labwc.github.io/labwc-actions.5.html#entry_action_name=setdecorations_decorations=value_forcessd=no) instead of ToggleDecorations.

Thanks for the hint. I already forgot that this was a thing in the 0.7.3 release cycle >.>
Will experiment with that

**Reply by Consolatis** [reply]
> Except that with SSD `sendtodesktop` and `toggledecorations` are working, but when I click "send link by email" in firefox it switches the desktop and activates TB.

If it opens a new window that might indeed cause it to a) get focus and thus b) switch workspaces. This is not actually a focus request but labwc giving a new window focus. I am not sure if child windows spawn on the parent's workspace or the current one though, based on your observation it seems to spawn on the parent's workspace.

Preventing focus for newly spawned windows based on window rules is something we thought about when implementing the ignore-focus-request feature but its not yet done.

**Reply by Narrat** [reply]
`SetDecorations` works perfectly. Thank you for reminding me of it, Consolatis.
And regarding the `FocusRequest`: Indeed, it seems to be the case of new windows (pw input, new e-mail)
